performance/build.gradle

```diff
@@ -17,140 +17,46 @@ buildscript {
-            project.configurations.getByName(kotlin.targets.jvm.compilations.main.runtimeDependencyConfigurationName)
-    )
-    classpath runtimeClasspath
-    main = "MainKt"
-    args "$jvmWarmup", "$attempts",  "${buildDir.absolutePath}/${jvmBenchResults}"
-}
+defaultTasks 'konanRun'
 
-task konanJsonReport {
-    doLast {
-        def nativeExecutable = MPPTools.getKotlinNativeExecutable(kotlin.targets.native, "RELEASE")
-        def nativeCompileTime = MPPTools.getNativeCompileTime(applicationName)
-        String benchContents = new File("${buildDir.absolutePath}/${nativeBenchResults}").text
-        def properties = getCommonProperties() + ['type'           : 'native',
-                                                  'compilerVersion': "${konanVersion}".toString(),
-                                                  'flags'          : kotlin.targets.native.compilations.main.extraOpts.collect{ "\"$it\"" },
-                                                  'benchmarks'     : benchContents,
-                                                  'compileTime'    : nativeCompileTime,
-                                                  'codeSize'       : MPPTools.getCodeSizeBenchmark(applicationName, nativeExecutable) ]
-        def output = MPPTools.createJsonReport(properties)
-        new File("${buildDir.absolutePath}/${nativeJson}").write(output)
-        uploadBenchmarkResultToBintray(nativeJson)
-    }
-}
-
-task jvmJsonReport {
-    doLast {
-        def jarPath = project.getTasks().getByName("jvmJar").archivePath
-        def jvmCompileTime = MPPTools.getJvmCompileTime(applicationName)
-        String benchContents = new File("${buildDir.absolutePath}/${jvmBenchResults}").text
-        def properties = getCommonProperties() + ['type'           : 'jvm',
-                                                  'compilerVersion': "${buildKotlinVersion}".toString(),
-                                                  'benchmarks'     : benchContents,
-                                                  'compileTime'    : jvmCompileTime,
-                                                  'codeSize'       : MPPTools.getCodeSizeBenchmark(applicationName, "${jarPath}") ]
-        def output = MPPTools.createJsonReport(properties)
-        new File("${buildDir.absolutePath}/${jvmJson}").write(output)
-        uploadBenchmarkResultToBintray(jvmJson)
-    }
-}
-
-jvmRun.finalizedBy jvmJsonReport
-konanRun.finalizedBy konanJsonReport
-
-private void dumpReport(String name, ByteArrayOutputStream output) {
-    new File("${name}").withOutputStream {
-        it.write(output.toByteArray())
-    }
+// Produce and send slack report.
+task slackReport(type: RegressionsReporter) {
+    // Create folder for report (root Kotlin project settings make create report in separate folder).
+    def reportDirectory = outputReport.substring(0, outputReport.lastIndexOf("/"))
+    mkdir reportDirectory
+    mkdir "../targetsResults"
```

Comment:
Same here, better use platform provided mechanisms of getting parent directories and such.